auth/key_manager.go

```diff
@@ -5,46 +5,53 @@ package auth
 
 import (
 	"errors"
-
-	"github.com/lestrrat-go/jwx/v2/jwk"
-	"github.com/lestrrat-go/jwx/v2/jwt"
 )
 
 var (
 	ErrUnsupportedKeyAlgorithm = errors.New("unsupported key algorithm")
 	ErrInvalidSymmetricKey     = errors.New("invalid symmetric key")
+	ErrPublicKeysNotSupported  = errors.New("public keys not supported for symmetric algorithm")
 )
 
-// JWK represents a JSON Web Key.
-type JWK struct {
-	key jwk.Key
-}
+// PublicKeyInfo represents a public key for external distribution via JWKS.
+// This follows RFC 7517 (JSON Web Key) specification.
+type PublicKeyInfo struct {
+	KeyID     string `json:"kid"`
+	KeyType   string `json:"kty"`
+	Algorithm string `json:"alg"`
+	Use       string `json:"use,omitempty"`
 
-// NewJWK creates a new JWK from a jwk.Key.
-func NewJWK(key jwk.Key) JWK {
-	return JWK{key: key}
-}
+	// EdDSA (Ed25519) fields
+	Curve string `json:"crv,omitempty"`
```

Comment:
Why don't we have separate object for this? 
Why this should be in struct that going to be returned in Public ? 
Is it safe to expose these information in public ? 
Is this needed for Public JWKS